samples/tables.js

```diff
@@ -304,6 +304,301 @@ function loadFileFromGCS(datasetId, tableId, bucketName, filename, projectId) {
   // [END bigquery_load_from_gcs]
 }
 
+function loadCSVFromGCS(datasetId, tableId, projectId) {
+  // [START bigquery_load_table_gcs_csv]
+  // Imports the Google Cloud client libraries
+  const BigQuery = require('@google-cloud/bigquery');
+  const Storage = require('@google-cloud/storage');
+
+  /**
+   * TODO(developer): Uncomment the following lines before running the sample.
+   */
+  // const projectId = "your-project-id";
+  // const datasetId = "my_dataset";
+  // const tableId = "my_table";
+
+  /**
+   * This sample loads the CSV file at
+   * https://storage.googleapis.com/cloud-samples-data/bigquery/us-states/us-states.json
+   *
+   * TODO(developer): Replace the following lines with the path to your file.
+   */
+  const bucketName = "cloud-samples-data";
+  const filename = "bigquery/us-states/us-states.csv";
+
+  // Instantiates clients
+  const bigquery = new BigQuery({
+    projectId: projectId,
+  });
+
+  const storage = Storage({
+    projectId: projectId,
+  });
+
+  // Configure the load job. For full list of options, see:
+  // https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.load
+  const metadata = {
+    'sourceFormat': 'CSV',
+    'skipLeadingRows': 1,
+    'schema': {
+      'fields': [
+        {'name': 'name', 'type': 'STRING'},
+        {'name': 'post_abbr', 'type': 'STRING'}
+      ]
+    }
+  };
+
+  let job;
+
+  // Loads data from a Google Cloud Storage file into the table
+  bigquery
+    .dataset(datasetId)
+    .table(tableId)
+    .load(storage.bucket(bucketName).file(filename), metadata)
+    .then(results => {
+      job = results[0];
+      console.log(`Job ${job.id} started.`);
+
+      // Wait for the job to finish
+      return job;
```

Comment:
> `return job;`

This doesn't seem right to me, since when has just `return job;` worked? Looking at https://github.com/googleapis/nodejs-common/blob/master/src/operation.js#L94, it would seem this should be `return job.promise;`. That would mean the existing bigquery samples in this file are wrong too.

Reply:
Seems to make the tests actually wait for the job now, but now they are failing for another reason.

```
$ node tables.js load-gcs-csv swast-scratch test_dataset us_states_csv
Job swast-scratch:US.5d1c5cc9-1ac1-436e-8d14-3c59f60a58fe started.
ERROR: TypeError: Cannot read property 'status' of undefined
    at bigquery.dataset.table.load.then.then.metadata (/Users/swast/src/nodejs-bigquery/samples/tables.js:367:31)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
```

Based on your link, it does seem the metadata is supposed to be passed in from the promise, but apparently it's not.

Reply:
I think we need to return `job.promise()`

Reply:
Hmm... `ERROR: TypeError: job.promise is not a function`

Reply:
Woops, looks like `load()` doesn't return an operation or Job object at all. When the callback or promise is resolved, the job is complete: https://github.com/googleapis/nodejs-bigquery/blob/fbe4f93f9167e904ee9d0beaf27f0d0b9da6d7bb/src/table.js#L1688

Reply:
Good catch @stephenplusplus ! I've mailed https://github.com/googleapis/nodejs-bigquery/pull/67 to fix the existing samples.